src/v/lsm/io/disk_persistence.cc

```diff
@@ -0,0 +1,352 @@
+              std::move(filepath), std::move(stream));
+            co_return ptr;
+        } catch (const std::system_error& e) {
+            if (e.code() == std::errc::no_such_file_or_directory) {
+                co_return std::nullopt;
+            }
+            throw io_error_exception(
+              e.code(), "io error opening seq file reader: {}", e);
+        } catch (...) {
+            throw io_error_exception(
+              "io error opening seq file reader: {}", std::current_exception());
+        }
+    }
+
+    ss::future<optional_pointer<random_access_file_reader>>
+    open_random_access_reader(std::string_view name) override {
+        try {
+            auto filepath = path(name);
+            auto file = co_await ss::open_file_dma(
+              filepath.native(), ss::open_flags::ro);
+            std::unique_ptr<random_access_file_reader> ptr;
+            ptr = std::make_unique<disk_file_reader>(
+              std::move(filepath), std::move(file));
+            co_return ptr;
+        } catch (const std::system_error& e) {
+            if (e.code() == std::errc::no_such_file_or_directory) {
+                co_return std::nullopt;
+            }
+            throw io_error_exception(
+              e.code(), "io error opening file reader: {}", e);
+        } catch (...) {
+            throw io_error_exception(
+              "io error opening file reader: {}", std::current_exception());
+        }
+    }
+
+    ss::future<std::unique_ptr<sequential_file_writer>>
+    open_sequential_writer(std::string_view name) override {
+        try {
+            auto filepath = path(name);
+            auto file = ss::open_file_dma(
+              filepath.native(),
+              ss::open_flags::create | ss::open_flags::rw
+                | ss::open_flags::truncate);
+            auto stream = co_await ss::with_file_close_on_failure(
+              std::move(file), [](ss::file& f) {
+                  return ss::make_file_output_stream(
+                    std::move(f), ss::file_output_stream_options{});
+              });
+            co_return std::make_unique<disk_seq_file_writer>(
+              std::move(filepath), std::move(stream));
+        } catch (const std::system_error& e) {
+            throw io_error_exception(
+              e.code(), "io error opening file writer: {}", e);
+        } catch (...) {
+            throw io_error_exception(
+              "io error opening file writer: {}", std::current_exception());
+        }
+    }
+
+    ss::future<> write_file_atomically(
+      std::string_view name, std::string_view contents) override {
+        auto staging_name = fmt::format(
+          "{}.{}.lsm-staging", name, uuid_t::create());
+        auto writer = co_await open_sequential_writer(staging_name);
+        auto future = co_await ss::coroutine::as_future<>(
+          writer->append(iobuf::from(contents)));
+        co_await writer->close();
+        if (future.failed()) {
```

Comment:
nit: if `future` is failed and `writer->close` the `future` is an abandoned exceptional future

Reply:
This code was removed in c75260b287